community_tasks/swiss_legal_evals.py

```diff
@@ -0,0 +1,476 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# ruff: noqa: F405, F403, F401
+"""
+This module contains task configurations and prompt functions for evaluating
+LLM models on Swiss legal datasets. Each task is defined using the 
+`LightevalTaskConfig` class with its respective prompt function. The tasks 
+cover a variety of benchmarks, including: translation of laws, court decisions 
+and press releases.
+
+Author: Joel Niklaus
+"""
+
+import statistics
+from dataclasses import dataclass
+from packaging import version
+import importlib.metadata as importlib_metadata
+
+import nltk
+from nltk import word_tokenize
+from nltk.translate import meteor_score
+
+from comet import download_model, load_from_checkpoint
+
+from transformers import AutoTokenizer, AutoModelForSequenceClassification
+from lighteval.logging.hierarchical_logger import hlog_warn
+from lighteval.tasks.lighteval_task import LightevalTaskConfig
+from lighteval.tasks.requests import Doc
+from lighteval.metrics.metrics import Metrics
+from lighteval.metrics.metrics_sample import JudgeLLMMixEval, BertScore
+from lighteval.metrics.utils.metric_utils import (
+    MetricCategory,
+    MetricUseCase,
+    SampleLevelMetric,
+    SampleLevelMetricGrouping,
+)
+from lighteval.metrics.imports.bert_scorer import BERTScorer
+from lighteval.metrics.normalizations import remove_braces, remove_braces_and_strip
+from lighteval.tasks.extended.mix_eval.main import process_judge_response_freeform_gpt
+from lighteval.tasks.extended.mix_eval.judge_prompts import (
+    gpt_judge_for_closeended_freeform,
+)
+
+
+# CUSTOM METRICS
+# TODO: adjust prompt
+
+
+def freeform_gpt_judge(judge_model_name: str = "gpt-4o"):
+    return SampleLevelMetricGrouping(
+        metric_name=[f"llm_judge_{judge_model_name}"],
+        higher_is_better={"judge_score_{judge_model_name}": True},
+        category=MetricCategory.LLM_AS_JUDGE,
+        use_case=MetricUseCase.SUMMARIZATION,
+        sample_level_fn=JudgeLLMMixEval(
+            judge_model_name=judge_model_name,
+            template=gpt_judge_for_closeended_freeform,
+            process_judge_response=process_judge_response_freeform_gpt,
```

Comment:
Hi @JoelNiklaus ! Have you taken a look at how those were implemented ? They are written for mix_eval mainly, they could be suited to your needs but you probably need to write your own judge  prompte template (a simple function with the same signature as the one you are using now)

Reply:
Yes, I am currently looking into adapting the prompt.